src/crimson/osd/main.cc

```diff
@@ -0,0 +1,103 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:nil -*-
+// vim: ts=8 sw=2 smarttab
+
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <iostream>
+
+#include <seastar/core/app-template.hh>
+#include <seastar/core/thread.hh>
+
+#include "common/ceph_argparse.h"
+#include "crimson/common/config_proxy.h"
+#include "crimson/net/SocketMessenger.h"
+
+#include "osd.h"
+
+using config_t = ceph::common::ConfigProxy;
+
+void usage(const char* prog) {
+  std::cout << "usage: " << prog << " -i <ID>" << std::endl;
+  generic_server_usage();
+}
+
+int main(int argc, char* argv[])
+{
+  std::vector<const char*> args{argv + 1, argv + argc};
+  if (ceph_argparse_need_usage(args)) {
+    usage(argv[0]);
+    return EXIT_SUCCESS;
+  }
+  std::string cluster;
+  std::string conf_file_list;
+  // ceph_argparse_early_args() could _exit(), while local_conf() won't ready
+  // until it's started. so do the boilerplate-settings parsing here.
+  auto init_params = ceph_argparse_early_args(args,
+                                              CEPH_ENTITY_TYPE_OSD,
+                                              &cluster,
+                                              &conf_file_list);
+  // TODO: add heartbeat
+  seastar::app_template app;
+  // talk with osd
+  ceph::net::SocketMessenger cluster_msgr("cluster", getpid());
+  // talk with mon/mgr
+  ceph::net::SocketMessenger client_msgr("client", getpid());
+  seastar::sharded<OSD> osd;
```

Comment:
We're using single shard at the moment. Theoretically this construct can be easily exchanged with just `OSD osd;` IIRC.

Reply:
we can. this prototyping was written when it's not decided that we want to go with 1:1 mapping for testing. IMO, switching over to `OSD osd;` does not do us any good, if `seastar::sharded<>` does not complicate the code or hurt the performance. moreover, we need to have a co-located OSD in the same process, if the 1:1 mapping profiling goes well. by then, we still need to have a `seastar::sharded<OSD>`.